controllers/metal3remediation_controller_test.go

```diff
@@ -190,230 +224,373 @@ func setReconcileNormalRemediationExpectations(ctrl *gomock.Controller,
 }
 
 var _ = Describe("Metal3Remediation controller", func() {
-	Describe(
-		"Test MachineRemediationReconcileNormal",
-		func() {
-			var ctrl *gomock.Controller
-			var remReconcile *Metal3RemediationReconciler
-
-			BeforeEach(func() {
-				ctrl = gomock.NewController(GinkgoT())
-				fakeClient := fake.NewClientBuilder().WithScheme(setupScheme()).Build()
-
-				remReconcile = &Metal3RemediationReconciler{
-					Client:         fakeClient,
-					ManagerFactory: baremetal.NewManagerFactory(fakeClient),
-					Log:            logr.Discard(),
+	var goMockCtrl *gomock.Controller
+	var testReconciler *Metal3RemediationReconciler
+	var fakeClient client.WithWatch
+	var objects []client.Object
+
+	defaultTestRequest := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      metal3RemediationName,
+			Namespace: namespaceName,
+		},
+	}
+	defaultCluster := newCluster(clusterName, nil, nil)
+
+	BeforeEach(func() {
+		goMockCtrl = gomock.NewController(GinkgoT())
+	})
+
+	AfterEach(func() {
+		goMockCtrl.Finish()
+	})
+
+	DescribeTable("Metal3Remediation Reconcile test",
+		func(tc reconcileRemediationTestCase) {
+			objects = []client.Object{}
+
+			if tc.Metal3RemediationCantBeFound {
+				fakeClient = fake.NewClientBuilder().WithScheme(setupScheme()).Build()
+			} else {
+				objects = append(objects, defaultCluster)
+				if tc.Metal3Remediation != nil {
+					objects = append(objects, tc.Metal3Remediation)
 				}
-			})
+				if tc.Machine != nil {
+					objects = append(objects, tc.Machine)
+				}
+				if tc.M3Machine != nil {
+					objects = append(objects, tc.M3Machine)
+				}
+				fakeClient = fake.NewClientBuilder().WithScheme(setupScheme()).WithObjects(objects...).Build()
+			}
+			testReconciler = &Metal3RemediationReconciler{
+				Client:         fakeClient,
+				ManagerFactory: baremetal.NewManagerFactory(fakeClient),
+				Log:            logr.Discard(),
+			}
+			_, err := testReconciler.Reconcile(context.TODO(), tc.TestRequest)
+			evaluateTestError(tc.ExpectedError, err)
+		},
+		Entry("Metal3Remediation haven't been found",
+			reconcileRemediationTestCase{
+				TestRequest:                  defaultTestRequest,
+				ExpectedError:                nil,
+				Metal3RemediationCantBeFound: true,
+			}),
+		Entry("Error retrieving owner machine",
+			reconcileRemediationTestCase{
+				TestRequest:                      defaultTestRequest,
+				ExpectedError:                    &ownerMachineErrorMsg,
+				FailedToCreateRemediationManager: true,
+				Metal3Remediation: &infrav1.Metal3Remediation{ObjectMeta: metav1.ObjectMeta{
+					Name:      metal3RemediationName,
+					Namespace: namespaceName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "Machine",
+							Name:       "wrongName",
+						},
+					},
+				}},
+				Machine: newMachine(clusterName, machineName, metal3machineName, "mynode"),
+			}),
+		Entry("Owner machine not set",
```

Comment:
This message sounds confusing to me, if you meant, ownerRef kind is wrong, let's be explicit here